File: patch_app.py
with open('gui/frontend/src/App.tsx', 'r') as f:
    content = f.read()

# Insert the import
import_str = "import SystemOverview from './components/features/SystemOverview'\n"
content = content.replace("import WhaleConstellations from './components/features/WhaleConstellations'",
                          "import WhaleConstellations from './components/features/WhaleConstellations'\n" + import_str)

# Insert the component at the very beginning of the main-grid
component_str = "\n        {/* Feature 0: System Overview (Spans full width) */}\n        <SystemOverview />\n"
content = content.replace("<div className=\"main-grid\"", "<div className=\"main-grid\"") # just to find it safely
content = content.replace("{/* Feature 1: Arbitrage Wormhole (Spans full width) */}",
                          component_str + "\n        {/* Feature 1: Arbitrage Wormhole (Spans full width) */}")

with open('gui/frontend/src/App.tsx', 'w') as f:
    f.write(content)

File: patch_app.js
const fs = require('fs');

const appFile = 'gui/frontend/src/App.tsx';
let content = fs.readFileSync(appFile, 'utf8');

const imports = `
import DarkPoolSonar from './components/features/DarkPoolSonar'
import FlashCrashMatrix from './components/features/FlashCrashMatrix'
import GalaxyView from './components/features/GalaxyView'
import SentimentWordCloud from './components/features/SentimentWordCloud'
import GasHologram from './components/features/GasHologram'
`;

content = content.replace("import OrbitalPortfolio from './components/features/OrbitalPortfolio';", "import OrbitalPortfolio from './components/features/OrbitalPortfolio';" + imports);

const featuresCode = `
        {/* NEW WORLD CLASS FEATURES */}
        <DarkPoolSonar />
        <FlashCrashMatrix />
        <GalaxyView />
        <SentimentWordCloud />
        <GasHologram />
`;

content = content.replace("        <ArbitrageWormhole />", featuresCode + "\n        <ArbitrageWormhole />");

fs.writeFileSync(appFile, content);
